code/datums/recipe.dm

```diff
@@ -1,102 +1,87 @@
 
-/datum/recipe/proc/check_reagents(var/datum/reagents/avail_reagents)
-	. = 1
-	for (var/r_r in reagents)
-		var/aval_r_amnt = REAGENT_VOLUME(avail_reagents, r_r)
-		if (!(abs(aval_r_amnt - reagents[r_r])<0.5)) //if NOT equals
-			if (aval_r_amnt>reagents[r_r])
-				. = 0
-			else
-				return -1
-	if (length(reagents) < LAZYLEN(avail_reagents.reagent_volumes))
-		return 0
-	return .
+/decl/recipe/Initialize()
+	. = ..()
+	global.microwave_recipes += src
+	for(var/thing in reagents)
+		global.microwave_accepts_reagents |= thing
+	for(var/thing in items)
+		global.microwave_accepts_items |= thing
+	global.microwave_maximum_item_storage = max(global.microwave_maximum_item_storage, length(items))
 
-/datum/recipe/proc/check_fruit(var/obj/container)
-	. = 1
-	if(fruit && fruit.len)
-		var/list/checklist = list()
-		 // You should trust Copy().
-		checklist = fruit.Copy()
-		for(var/obj/item/chems/food/snacks/grown/G in container)
-			if(!G.seed || !G.seed.kitchen_tag || isnull(checklist[G.seed.kitchen_tag]))
-				continue
-			checklist[G.seed.kitchen_tag]--
-		for(var/ktag in checklist)
-			if(!isnull(checklist[ktag]))
-				if(checklist[ktag] < 0)
-					. = 0
-				else if(checklist[ktag] > 0)
-					. = -1
-					break
-	return .
+/decl/recipe/proc/check_reagents(var/datum/reagents/avail_reagents)
+	if(length(avail_reagents?.reagent_volumes) < length(reagents))
+		return FALSE
+	for(var/rtype in reagents)
+		if(REAGENT_VOLUME(avail_reagents, rtype) < reagents[rtype])
+			return FALSE 
+	return TRUE
 
-/datum/recipe/proc/check_items(var/obj/container)
-	. = 1
-	if (items && items.len)
-		var/list/checklist = list()
-		checklist = items.Copy() // You should really trust Copy
-		for(var/obj/O in container.get_contained_external_atoms())
-			if(istype(O,/obj/item/chems/food/snacks/grown))
-				continue // Fruit is handled in check_fruit().
-			var/found = 0
-			for(var/i = 1; i < checklist.len+1; i++)
-				var/item_type = checklist[i]
-				if (istype(O,item_type))
-					checklist.Cut(i, i+1)
-					found = 1
-					break
-			if (!found)
-				. = 0
-		if (checklist.len)
-			. = -1
-	return .
+/decl/recipe/proc/check_fruit(var/obj/container)
+	if(!length(fruit))
+		return TRUE
+	var/container_contents = container?.get_contained_external_atoms()
+	if(length(container_contents) < length(fruit))
+		return FALSE
+	var/list/needed_fruits = fruit.Copy()
+	for(var/obj/item/chems/food/snacks/grown/G in container_contents)
+		var/ktag = G.seed?.kitchen_tag
+		if(needed_fruits[ktag] > 0)
+			needed_fruits[ktag]--
+	for(var/ktag in needed_fruits)
+		if(needed_fruits[ktag] > 0)
+			return FALSE
+	return TRUE
+
+/decl/recipe/proc/check_items(var/obj/container)
+	if(!length(items))
+		return TRUE
+	var/container_contents = container?.get_contained_external_atoms()
+	if(length(container_contents) < length(items))
+		return FALSE
+	var/list/needed_items = items.Copy()
+	for(var/itype in needed_items)
+		for(var/atom/movable/AM as anything in container_contents)
```

Comment:
Why use a typed var at all if you're just doing an istype?

Reply:
Avoiding a cast.

Reply:
Nvm, just realized type is irrelevant there.